pkg/controller/runnable/reconciler.go

```diff
@@ -38,26 +41,47 @@ type Reconciler struct {
 	DynamicTracker          tracker.DynamicTracker
 	ConditionManagerBuilder conditions.ConditionManagerBuilder
 	conditionManager        conditions.ConditionManager
+	RepositoryBuilder       repository.RepositoryBuilder
+	ClientBuilder           realizerclient.ClientBuilder
+	RunnableCache           repository.RepoCache
+	logger                  logr.Logger
 }
 
 func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	logger := logr.FromContext(ctx)
-	logger.Info("started")
-	defer logger.Info("finished")
+	r.logger = logr.FromContext(ctx)
```

Comment:
next PR is removing logger from struct, but I think that is okay if we don't deal with this here.